dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -836,8 +836,16 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
                                                       jpeg_streams_[jpeg_stream_idx], stream),
                      file_name);
 
-      NVJPEG_CALL_EX(nvjpegDecodeJpegDevice(handle_, decoder, state, &nvjpeg_image, stream),
-                     file_name);
+      ret = nvjpegDecodeJpegDevice(handle_, decoder, state, &nvjpeg_image, stream);
+      // if nvJPEG fails try HostDecoder
+      if (ret != NVJPEG_STATUS_SUCCESS) {
+        auto warning_msg = make_string("NVJPEG error \"", static_cast<int>(ret), "\" : ",
+                           nvjpeg_parse_error_code(ret), " ", file_name);
+        DALI_WARN(warning_msg);
```

Comment:
I'm not sure about this, but maybe we should warn as fallback to CPU at this stage is not expected and indicates a problem with nvJPEG we should see and not silently ignore.